src/network.js

```diff
@@ -1,51 +1,139 @@
+import FingerprintJS from '@fingerprintjs/fingerprintjs'
 const axios = require('axios');
 
 
 class Network {
     constructor(featureUrl, apiUrl = 'http://localhost:8080') {
         this.url = apiUrl + '/api/v1';
-        this.featureUrl = featureUrl
+        this.featureUrl = featureUrl;
+        this.accessToken = null;
+        // Initialize fingerprint
+        this.fpPromise = FingerprintJS.load({monitoring: false});
+    }
+
+    /**
+     * Execute the request. If it fails with 401 Unauthorized, fetch an access token and retry.
+     * 
+     * @param {function} requestPromise Function returning a promise which executes the request
+     * @returns the request's result
+     */
+    #sendRequestWithOAuth(requestPromise, ...args) {
+        return new Promise((resolve, reject) => {
+            let retried = false;
+            
+            const executePromise = async () => {
+                requestPromise(...args)
+                .then(resolve)
+                .catch(async (error) => {
+                    if (!retried && error.response.status == 401) {
+                        try {
+                            const authResponse = await axios.post(this.url + '/authorize', {
+                                withCredentials: true,
+                                headers: {
+                                    'Access-Control-Allow-Origin': '*',
+                                    'Content-Type': 'application/json'
+                                },
+                            });
+                            const tokenResponse = await axios.post(this.url + '/token', {
+                                grant_type: 'authorization_code',
+                                code: authResponse.data.code
+                            },
+                            {
+                                withCredentials: false,
+                                headers: {
+                                    'Access-Control-Allow-Origin': '*',
+                                    'Content-Type': 'application/json'
+                                },
+                            })
+                            this.accessToken = tokenResponse.data.access_token;
+                            retried = true;
+                            executePromise();
+                        } catch(error) {
+                            reject(error);
+                        }
+                    } else {
+                        reject(error);
+                    }
+                });
+            }
+            executePromise();
+        })
+    }
+
+    #checkDisplayPromise() {
+        return new Promise((resolve, reject) => {
+            axios.get(this.url + '/rules', {
```

Comment:
To improve readability, you can use await instead of using .then(), and you can use a try-catch block instead of using .catch().